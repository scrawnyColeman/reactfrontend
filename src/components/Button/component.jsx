import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { default as MaterialButton } from '@material-ui/core/Button';
import { colours, materialStyles } from '../../constants/styles';

const Button = ({ text, hierarchy = 'primary', size = 'large', onClick }) => {
    const useStyles = makeStyles(materialStyles);
    const classes = useStyles();
    const buttonOverrideStyles = {
        fontSize: size === 'large' ? '1.25rem' : '1rem',
        textTransform: 'unset',
        color: colours.primary,
    };
    return (
        <MaterialButton
            color={hierarchy}
            size={size}
            className={classes.button}
            style={buttonOverrideStyles}
            variant="text"
            onClick={onClick}
        >
            {text}
        </MaterialButton>
    );
};

export default Button;
