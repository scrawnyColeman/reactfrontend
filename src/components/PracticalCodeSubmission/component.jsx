import React from 'react';
import styled from 'styled-components';
import { colours } from '../../constants/styles';
import TextEditor from '../TextEditor/component';
import Button from '../Button/component';

const Container = styled.div`
    height: 65vh;
    border: 2px solid ${colours.primary};
    padding: 16px;
    display: grid;
    grid-template-rows: min-content 1fr;
    overflox-y: scroll;
`;

const StyledButtonContainer = styled.div`
    margin-top: 24px;
    text-align: right;
`;

const PracticalCodeSubmission = ({ data, getHelp, toggleSubmission, submitted }) => (
    <Container>
        <TextEditor
            mode={data.language}
            readOnly={false}
            minLines={20}
            maxLines={20}
            placeholder={`// Write your code here\n//Don't be afraid to seek help from the forum`}
            fontSize={18}
        />
        <StyledButtonContainer>
            <Button text="Get help" onClick={getHelp} />
            <Button text={submitted ? 'View Question' : 'View Solution'} onClick={toggleSubmission} />
        </StyledButtonContainer>
    </Container>
);

export default PracticalCodeSubmission;
