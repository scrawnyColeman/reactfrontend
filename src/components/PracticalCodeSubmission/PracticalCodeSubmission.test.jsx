import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import PracticalCodeSubmission from './component';
import Button from '../Button/component';
Enzyme.configure({ adapter: new Adapter() });

describe('PracticalCodeSubmission component', () => {
    const getHelp = jest.fn();
    const toggleSubmission = jest.fn();
    const submitted = false;
    const data = { language: 'Java' };

    test('component matches snapshot', () => {
        const wrapper = shallow(
            <PracticalCodeSubmission
                getHelp={getHelp}
                toggleSubmission={toggleSubmission}
                submitted={submitted}
                data={data}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('getHelp prop called on Button click', () => {
        const wrapper = shallow(
            <PracticalCodeSubmission
                getHelp={getHelp}
                toggleSubmission={toggleSubmission}
                submitted={submitted}
                data={data}
            />,
        );
        wrapper.find(Button).at(0).simulate('click');

        expect(getHelp).toHaveBeenCalledTimes(1);
    });

    test('toggleSubmission prop called on Button click', () => {
        const wrapper = shallow(
            <PracticalCodeSubmission
                getHelp={getHelp}
                toggleSubmission={toggleSubmission}
                submitted={submitted}
                data={data}
            />,
        );
        wrapper.find(Button).at(1).simulate('click');

        expect(toggleSubmission).toHaveBeenCalledTimes(1);
    });

    test.each`
        Text               | Submitted
        ${'View Question'} | ${true}
        ${'View Solution'} | ${false}
    `('Toggle Button has correct text when submission is $Submitted', ({ Text, Submitted }) => {
        const wrapper = shallow(
            <PracticalCodeSubmission
                getHelp={getHelp}
                toggleSubmission={toggleSubmission}
                submitted={Submitted}
                data={data}
            />,
        );
        const mainButton = wrapper.find(Button).at(1);

        expect(mainButton.props().text).toBe(Text);
    });
});
