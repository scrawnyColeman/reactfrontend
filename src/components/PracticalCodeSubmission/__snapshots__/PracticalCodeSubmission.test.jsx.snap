// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PracticalCodeSubmission component component matches snapshot 1`] = `
<styled.div>
  <TextEditor
    fontSize={18}
    maxLines={20}
    minLines={20}
    mode="Java"
    placeholder="// Write your code here
//Don't be afraid to seek help from the forum"
    readOnly={false}
  />
  <styled.div>
    <Button
      onClick={[MockFunction]}
      text="Get help"
    />
    <Button
      onClick={[MockFunction]}
      text="View Solution"
    />
  </styled.div>
</styled.div>
`;
